android/expoview/src/main/java/host/exp/exponent/notifications/managers/SchedulerManager.java

```diff
@@ -0,0 +1,150 @@
+package host.exp.exponent.notifications.managers;
+
+import android.content.Context;
+import com.raizlabs.android.dbflow.sql.language.Select;
+import org.unimodules.core.interfaces.Function;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import host.exp.exponent.notifications.exceptions.UnableToScheduleException;
+import host.exp.exponent.notifications.interfaces.SchedulerInterface;
+import host.exp.exponent.notifications.interfaces.SchedulersManagerInterface;
+import host.exp.exponent.notifications.schedulers.CalendarScheduler;
+import host.exp.exponent.notifications.schedulers.IntervalScheduler;
+
+class SchedulerManager implements SchedulersManagerInterface {
```

Comment:
If it was a kernel service we could have some configuration power over how it is initialized and we could pass in a list of supported schedulers — a possibility for expanding the supported types. (You don't have to implement it, it's just a remark)